internal/controller/managedcluster_controller.go

```diff
@@ -514,8 +514,24 @@ func (r *ManagedClusterReconciler) machinesAvailable(ctx context.Context, namesp
 	return len(itemsList.Items) != 0, nil
 }
 
+func ExtractTemplateName(rawObj client.Object) []string {
+	cluster, ok := rawObj.(*hmc.ManagedCluster)
+	if !ok {
+		return nil
+	}
+	if cluster.Spec.Template == "" {
+		return []string{}
+	}
+	return []string{cluster.Spec.Template}
+}
+
 // SetupWithManager sets up the controller with the Manager.
 func (r *ManagedClusterReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	if err := mgr.GetFieldIndexer().
```

Comment:
But just to confirm since I'm still in doubt. We should register the FieldIndexer not only for the ManagedCluster reconciler but also for the Webhook. In case, we register it here will it work in the admission webhook code?

Reply:
The indexer is registered for the manager, not for the reconciler, so it doesn't really matter where we do that. Logically it makes sense to keep it near the component that is supposed to use it (webhook in our case) or right after we set up the manager.

Reply:
Thanks for the suggestions. "Reuse" is a great argument. I reworked this functionality.